### PR TITLE
Mailing lists: Language specific lists for countries

### PR DESCRIPTION
**Pull request by reinhardt**
syslabcom/scrum#1621

### CODE REVIEW

File: src/osha/oira/client/browser/client.py [hunk 1]
```diff
@@ -81,12 +83,51 @@ def __call__(self):
 class MailingListsJson(BaseJson):
     """Mailing lists (countries and tools, in the future also sectors)"""
 
+    @property
+    @memoize
+    def client_path(self):
+        return "/".join(self.context.getPhysicalPath())
+
+    def _get_mailing_lists_for(self, brain):
```
Comment (thet) [inline src/osha/oira/client/browser/client.py:91]: Does it make sense to memoize this method also?

File: src/osha/oira/client/browser/client.py [hunk 2]
```diff
@@ -81,12 +83,51 @@ def __call__(self):
...
+
+        return [
+            self._get_entry(
+                "-".join((brain.getId, language)), f"{brain.Title} ({language})"
+            )
+            for language in languages
+        ]
+
     @property
     def results(self):
```
Comment (thet) [inline src/osha/oira/client/browser/client.py:122]: Does it make sense to memoize this?

File: src/osha/oira/client/browser/client.py [hunk 3]
```diff
@@ -230,13 +266,32 @@ def get_token(self):
...
-            .filter(NewsletterSubscription.zodb_path.in_(group_paths))
+            .filter(NewsletterSubscription.zodb_path.in_(other))
             .group_by(Account.loginname)
         )
-        return [s.loginname for s in subscribers]
+        subscribers.extend([s.loginname for s in other_subscribers])
+        return subscribers
 
     @property
     def results(self):
```
Comment (thet) [inline src/osha/oira/client/browser/client.py:297]: Should this be memoized? Maybe out of scope. But I've seen a `memoize` on a `getPhysicalPath`. The memoize logic is also quite a chunk of code where I'm **not sure** what is more expensive - memoizing everything vs memoizing selectively.

Whatever. As soon as this results method depends on request data (user session, etc) memoizing does not make sense at all.

Never mind.

**Comment by reinhardt** [comment]
Sending a newsletter probably doesn't happen often enough to make memoization/caching necessary.

**Pull request merged**